### PR TITLE
Multiplatform venv

### LINKED ISSUE #232: Virtual Environment testing and instructions
### Problem:
The README instructs users to install this Command Line tool via pip. 
That is not really appropriate for a tool. Python is a dependency of this tool, not a feature.
The point of this tool is to allow encryption form a Command Line. As such, the README should:

1. Not assume the user has python knowledge
2. Not install the library to the system OS, but to a virtual environment

### Solution:

Update the README to instruct users to install the library in a virtual environment. 

Add testing/CI actions that ensure these directions work.

### Out of scope:

Because this a tool, not a library, it should really be installed by an OS package manager (i.e: `brew`, `yum`, `apt-get`, etc.). 
But the CI/CD work for that level of support is significantly greater than just supporting  and testing the tool in a virtual environment. Linking instructions for setting up a virtual environment, and then having users install the tool against that virtual environment is an affordable improvement that will delight non-Python users AND follow best practices of not modifying the OS's Python.

### PR DESCRIPTION
**Pull request by phasenoisepon**
*Issue #, if available:* closes #232

*Description of changes:*
- Adds setup instructions for setting up the tool in a python virtual environment in README.rst
- Adds github workflow yaml for Venv specifc testing on MacOS/Linux/Windows environments

Comments:
- Windows specific sourcing has to be done to test Windows/Powershell instructions in the readme
- Not the happiest with the duplication of code in the workflow, might be able to split the detection into each run step, but then you would have to deal with PowerShell / Bash incompatibility [(default shell docs)](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#using-a-specific-shell)
- PR'ed for Hacktoberfest (so if this passes muster I'd appreciate it being tagged with `hacktoberfest-accepted`)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -55,10 +55,37 @@ Installation
...
+      $ pip install aws-encryption-sdk-cli
+
+   Windows (PowerShell):
+
+   .. code-block:: sh
+
+      > cd my_project
+      > python3 -m venv env
+      > .\env\Scripts\Activate.ps1
+      (env) > pip install aws-encryption-sdk-cli
```
[the remark this answers is not in the document]
Comment (phasenoisepon) [reply README.rst:86]: On a vanilla windows powershell install, activating the virtual environment by sourcing the venv will add the venv's name to your shell.

File: .github/workflows/install.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+          python -m venv env
+          source env/bin/activate
+          pip install --upgrade -r ci-requirements.txt
+        if: matrix.platform.os != 'windows-latest'
+      - name: Windows setup venv and package
+        run: |
+          python -m pip install --upgrade pip    
+          python -m venv env
+          .\env\Scripts\activate
+          pip install --upgrade -r ci-requirements.txt
```
[the remark this answers is not in the document]
Comment (phasenoisepon) [reply .github/workflows/install.yaml]: resolved this in https://github.com/aws/aws-encryption-sdk-cli/pull/233/commits/1f036ccc205c42ac3ba7a1260e20d9bbfcc9757c

The venv setup step should install the package from pip, but inside of the venv (hence activating the venv). Or do you want the venv to be activated and installed into in the `test` step?

File: README.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (texastony) [reply README.rst:86]: Groovy. Thanks!

**Comment by texastony** [comment]
@phasenoisepon I may have led you astray... I thought that the environmental variables for using the AWS KMS keys were already configured... but they clearly are not there. 

I think I will need to look into I can do for a local test that encrypts and decrypts without calling KMS, or ensuring that the environmental variables are set up correctly.

Sorry for leading you astray.

File: .github/workflows/install.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      - name: Windows run test
+        env:
+          TOXENV: ${{ matrix.category }}
+        run: |
+          .\env\Scripts\activate
+          (Get-Command python).Path
+          INPUT_TEXT="hello world"
+          CIPHERTEXT=$(./examples/bin/encrypt_command_line.sh $INPUT_TEXT $AWS_ENCRYPTION_SDK_PYTHON_INTEGRATION_TEST_AWS_KMS_KEY_ID)
+          PLAINTEXT=$(./examples/bin/decrypt_command_line.sh "$CIPHERTEXT" $AWS_ENCRYPTION_SDK_PYTHON_INTEGRATION_TEST_AWS_KMS_KEY_ID)
+          python3 -c "assert '$PLAINTEXT' == '$INPUT_TEXT'"
```
Comment (texastony) [inline .github/workflows/install.yaml:63]: suggestion: Let's leave the encryption testing to the CI actions in AWS CodeBuild
Let us just ensure that the library is installed by checking that the version command returns. 
```suggestion
    aws-encryption-cli --version
```

File: .github/workflows/install.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      - name: MacOS Linux run test
+        env:
+          TOXENV: ${{ matrix.category }}
+        run: |
+          source env/bin/activate
+          which python
+          INPUT_TEXT="hello world"
+          CIPHERTEXT=$(./examples/bin/encrypt_command_line.sh $INPUT_TEXT $AWS_ENCRYPTION_SDK_PYTHON_INTEGRATION_TEST_AWS_KMS_KEY_ID)
+          PLAINTEXT=$(./examples/bin/decrypt_command_line.sh "$CIPHERTEXT" $AWS_ENCRYPTION_SDK_PYTHON_INTEGRATION_TEST_AWS_KMS_KEY_ID)
+          python3 -c "assert '$PLAINTEXT' == '$INPUT_TEXT'"
```
Comment (texastony) [inline .github/workflows/install.yaml:74]: suggestion: Let's leave the encryption testing to the CI actions in AWS CodeBuild
Let us just ensure that the library is installed by checking that the version command returns.
```suggestion
          aws-encryption-cli --version
```

**Review by texastony — CHANGES_REQUESTED** [review]
I thought about it, and came to the conclusion that ensuring the library is installed can just be done via the `get version` flag. 

I am so sorry that I lead you astray earlier. I fell for the oldest trick: it works on my machine, so it will work on all machines. 

Ironically, that is exactly what your PR is preventing. By ensuring the install instructions are tested, we ensure that this works on all supported machines.
Thanks again for this.

**Comment by phasenoisepon** [comment]
@texastony when you're in office, approve the other workflows.

latest commits should give this the functionality you were looking for.

**Review by texastony — APPROVED** [review]

**Pull request merged**